### ISSUE TITLE
User (role: guest) missing scope (documents.write) Problem

### ISSUE THREAD

**Issue by martin-matj** [appwrite/appwrite#4647]
this is my code. first it runs well. but after many people visiting, i got error: **User (role: guest) missing scope (documents.write)**
if i create a new user, change the data. it runs well again. but it will get into problems soon again. about more than 1000 visits.
what did i miss?
`      const client = new Client();

      client
        .setEndpoint("http://myserver:8887/v1")
        .setProject("my-project");
      const account = new Account(client);
      const databases = new Databases(client, "my-database");

          const promise = account.createEmailSession(
            // "email@example.com",
            // "password"
          );

          promise.then(
            (response) => {
              const promise1 = databases.createDocument(
                "my-collection",
                "unique()",
                // json
                this.$data
              );

              promise1.then(
                (response) => {
                  console.log(response); // Success
                },
                (error) => {
                  });
                }
              );
            },
            (error) => {
              });
            }
          );
`

**Comment by stnguyen90** [comment]
Theres a limitation with the number of sessions a user can have...that said, are you hard coding the same email and password for every user who accesses your app?

**Comment by martin-matj** [comment]
yes, i guess this is the reason. but is there a configure to remove the limitation, or i'd better use anonymous session? Thanks!

**Comment by stnguyen90** [comment]
@martin-matj It doesn't quite make sense to hard code the same email and password for every user. Anonymous session probably makes more sense.

**Comment by martin-matj** [comment]
got it! thanks! @stnguyen90  I will try it!

**Comment by martin-matj** [comment]
@stnguyen90 i have tried. the anoymouse session better be deleted everytime after using it. otherwise it will cause the same error

**Comment by stnguyen90** [comment]
@martin-matj sessions are automatically persisted by the Appwrite sdks. So, what your app should do on start is check if there's a valid session already before trying to create a other one. You can check if there's a valid session by calling `account.get()`. If that throws an exception, there is no session and you can create a new anonymous session.

**Comment by martin-matj** [comment]
@stnguyen90 ok, i find the method, **account.get()**. but i wonder if there is also a limitation to anoymous session? i mean, too many people use the anonymous session, would it cause trouble?

**Comment by stnguyen90** [comment]
@martin-matj each Anonymous Session is essentially a User in Appwrite so you'll end up with a lot of Users. It's not really a problem...just maybe cluttered. You can create an Appwrite Function to delete inactive Anonymous Users if you want to keep it a little cleaner.

**Comment by martin-matj** [comment]
Ok， I think I understand. THANKS A LOT！ @stnguyen90
